pkg/kubectl/cmd/create.go

```diff
@@ -159,6 +167,14 @@ func RunCreate(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *re
 	return nil
 }
 
+func RunEditOnCreate(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, options *resource.FilenameOptions) error {
+	outputFlag := cmdutil.GetFlagString(cmd, "output")
+	if outputFlag != "json" && outputFlag != "yaml" {
+		return cmdutil.UsageError(cmd, "--output must be one of json|yaml, when --edit is true")
```

Comment:
If `--edit` is specified, we should set `--output` to either `json` or `yaml`, so that one can do `kubectl create --edit` without specifying `-o`. WDYT?